src/linter/block.go

```diff
@@ -2196,9 +2196,16 @@ func (b *BlockWalker) handleAssignReference(a *assign.Reference) bool {
 	return false
 }
 
-func (b *BlockWalker) handleAssignList(items []*expr.ArrayItem) {
-	for _, item := range items {
-		b.handleVariableNode(item.Val, meta.NewTypesMap("unknown_from_list"), "assign")
+func (b *BlockWalker) handleAssignList(items []*expr.ArrayItem, info meta.ClassInfo) {
+	for i, item := range items {
+		prop, ok := info.Properties[fmt.Sprint(i)]
+		var tp meta.TypesMap
+		if !ok {
+			tp = meta.NewTypesMap("unknown_from_list")
+		} else {
+			tp = prop.Typ
+		}
+		b.handleVariableNode(item.Val, tp, "assign")
```

Comment:
This method will handle all types like shape types. For non-shape types it's easier to keep the previous loop and avoid doing `fmt.Sprint` and map lookup per every item.